src/direct/lwt_direct.mli

```diff
@@ -0,0 +1,33 @@
+(** Direct style control flow for Lwt. *)
+
+val run : (unit -> 'a) -> 'a Lwt.t
+(** [run f] runs the function [f ()] in a task within
+    the [Lwt_unix] event loop. [f ()] can create [Lwt]
+    promises and use {!await} to wait for them. Like any promise
+    in Lwt, [f ()] can starve the event loop if it runs long computations
+    without yielding to the event loop.
+
+    When [f ()] terminates (successfully or not), the promise
+    [run f] is resolved with [f ()]'s result, or the exception
+    raised by [f ()]. *)
+
+val run_in_the_background :
+  ?on_uncaught_exn:(exn -> Printexc.raw_backtrace -> unit) ->
+  (unit -> unit) ->
+  unit
+(** [run_in_the_background f] is similar to [ignore (run f)].
+    The computation [f()] runs in the background in the event loop
+    and returns no result.
+    @param on_uncaught_exn if provided, this is called when [f()]
+    raises an exception. *)
+
+val yield : unit -> unit
+(** Yield to the event loop.
+    Can only be used inside {!run} or {!run_in_the_background}. *)
```

Comment:
```suggestion
    calling [yield] outside of {!run} or {!run_in_the_background} will raise an exception,
    crash your program, or otherwise cause errors. It is a programming error to do so. *)
```

Or something like that. I want it to be non-commital on what happens but also avoid "cannot do that" in documentation (it's never very clear).

Reply:
Maybe we can put `yield` after `await` and mention something along the lines of

> `yield` is similar to `await (Lwt.pause ())` but more efficient.